src/Init/System/IO.lean

```diff
@@ -329,34 +331,19 @@ def pathExists (p : FilePath) : IO Bool :=
 end System.FilePath
 
 namespace IO
-section
-variable [Monad m] [MonadLiftT IO m]
-
-def getStdin : m FS.Stream := liftM Prim.getStdin
-def getStdout : m FS.Stream := liftM Prim.getStdout
-def getStderr : m FS.Stream := liftM Prim.getStderr
-
-/-- Replaces the stdin stream of the current thread and returns its previous value. -/
-def setStdin : FS.Stream → m FS.Stream := liftM ∘ Prim.setStdin
-
-/-- Replaces the stdout stream of the current thread and returns its previous value. -/
-def setStdout : FS.Stream → m FS.Stream := liftM ∘ Prim.setStdout
-
-/-- Replaces the stderr stream of the current thread and returns its previous value. -/
-def setStderr : FS.Stream → m FS.Stream := liftM ∘ Prim.setStderr
 
-def withStdin [MonadFinally m] (h : FS.Stream) (x : m α) : m α := do
+def withStdin [MonadFinally IO] (h : FS.Stream) (x : IO α) : IO α := do
```

Comment:
The `[MonadFinally IO]` argument seems unnecessary now.  `withStdin` also seems more useful in the `m α → m α` version.